### ISSUE TITLE
Support For Multiple Lines?

### ISSUE THREAD

**Issue by charlespierce** [yaahc/displaydoc#16]
I was investigating using `displaydoc` to simplify our error enum in [Volta](https://volta.sh), however I quickly ran into the fact that it only supports the first line of any doc comments as the `Display` impl. All of our errors are intentionally multiple lines, so we can't currently take advantage of the simplification provided by `displaydoc`.

Would it make sense to support multiple lines of doc comments, or is there an explicit reason to only use the first line? I think I see the changes that would be needed to support multiple lines, and would be happy to open a PR to do that, but I don't want to break anything that was intentional.

**Comment by yaahc** [comment]
displaydoc does have support for multiple lines currently, though its not super intuitive and is more of an effect of how doc comments are represented as attributes. With `///` and `//!` each line is treated as its own `#[doc = "..."]` attribute, where as with block style doc comments like `/** */` the entire comment is treated as a single attribute, which lets you setup doc comments that span multiple lines.

There's an example of a multi-line doc comment in `tests/happy.rs` on variants 4 and 5

```rust
#[derive(Display)]
enum Happy {
    /// I really like Variant1
    Variant1,
    /// Variant2 is pretty swell 2
    Variant2,
    /// Variant3 is okay {sometimes}
    Variant3 { sometimes: &'static str },
    /**
     * Variant4 wants to have a lot of lines
     *
     * Lets see how this works out for it
     */
    Variant4,
    /// Variant5 just has {0} many problems
    /// but multi line comments aren't one of them
    Variant5(u32),

    /// The path {0}
    #[cfg(feature = "std")]
    Variant6(PathBuf),
}
```

Alternatively you can use `#[doc = "..."]` directly to get multiline doc comments.

So, I would prefer to not add support for multi line display methods via `///`, because that would prevent users from having doc comments that include more content than their display impl.

Are you okay with using a block doc comment? If so I think I'll add a section on multi line doc comments to the README and consider that the fix for this issue.

**Comment by charlespierce** [comment]
Ohh, I totally didn't know about block doc comments at all, that makes sense! That should fit our use case pretty easily, so yeah, a section in the REDME to make that more discoverable would be great.

Thanks!

**Comment by gibfahn** [comment]
Just ran into the same thing, TIL about multiple `///` comments!

Section in the README would be great for this, but as this is the top issue it's still fairly discoverable.

**Comment by yaahc** [comment]
resolved by #22
